React/props-state.md

```diff
@@ -0,0 +1,99 @@
+
+## `state`
+State is managed **within a given component**. State can be accessed using `this.state` and it can be modified using `this.setState()`.
+
+#### Overview
+- Can be assigned to any object key you want
+- `setState` will always take an object as the parameter
+- `setState` will merge any params passed in with the existing state data, it won't overwrite it
+- We often set the initial state in the constructor function
+
+A really important aspect of state is that whenever the state of a component is changed, it will be **updated**, calling the `render` function. For now, you can think of `setState` as doing the `setState` operation PLUS the `render` _automatically_. We'll learn even more about what else our component does when it is updated later on.
+
+#### Examples
+
+**Syntax**
+
+Commonly, initial state is set in a component's constructor function. This is set using variable assignment (the equals sign).
+```javascript
+constructor() {
+  super();
+  this.state = {
+    varNameOne: '',
+    varNameTwo: false
+  };
+}
+```
+
+Once the initial state is set, we can then make changes  using the `setState` function. (**Note:** `setState` will work even if that state variable was not included in the initial `state` object.) This function will merge this object with the existing state object, overriding any existing values on the same variables.
+
+```JavaScript
+this.setState({ varNameOne: 'value goes here'});
+```
+
+When you need to use `state`, you use `this.state.varNameOne` to access the value. (**Note:** You cannot set a state variable using the `=` directly)
+
+## With our powers combined...
+
+Oftentimes `props` and `state` can be used to support each other. If data is stored in a parent component in `state`, the parent component can render the child component by passing in the `state` value as a prop. Then, the child component will have access to the relevant data but not be concerned with how that data is managed.
+
+**Example**
+
+In the parent component, pass a state value into the child component as a prop when it is rendered.
+
+`<ComponentB myVariable={this.state.theVariable}`.
+
+Then, within the child component, access that value using `this.props.myVariable`.
+
+
+## Changing `props` and `state`
+Here is a helpful chart to assist you in determining whether data belongs in `props` or `state`. For any line item that has "Yes" for both, it means you need to make that decision based on the context of the problem you are trying to solve.
+
+ ?     | props     | state
+ :------------- | :------------- |:-------------
+Can get initial value from parent component?       | Yes   | Yes
+Can be changed by parent component? | Yes | No
+Can change inside component? | No | Yes
+Can set initial value for a child component? | Yes | Yes
+Can change in a child component? | Yes | No
+
```

Comment:
This material is great imo, especially because it stands alone. I definitely like the emphasis of "here's a lot of definition and you'll refer to these notes later"

It of course ties directly into the next lecture, but it definitely feels weird without the application. I wish there were a way to address this, though I can't think of any suggestions right now